File: src/main/java/com/itsz/portfolio/service/calculator/StockPriceCalculator.java
package com.itsz.portfolio.service.calculator;

import com.itsz.portfolio.entity.Security;
import org.springframework.stereotype.Component;

import java.math.BigDecimal;

@Component
final class StockPriceCalculator implements SecurityPriceCalculator {
    @Override
    public BigDecimal calculate(Security security, BigDecimal price) {
        return price;
    }
}

File: src/main/kotlin/com/itsz/portfolio/service/calculator/StockPriceCalculator.kt
package com.itsz.portfolio.service.calculator

import com.itsz.portfolio.entity.Security
import com.itsz.portfolio.entity.SecurityType
import org.springframework.stereotype.Component
import java.math.BigDecimal

@Component
internal class StockPriceCalculator : SecurityPriceCalculator {
    override fun calculate(security: Security, price: BigDecimal): BigDecimal = price
    override fun getSecurityType(): SecurityType = SecurityType.STOCK
}
